operator/pkg/helmreconciler/reconciler.go

```diff
@@ -660,7 +662,52 @@ func applyManifests(kubeClient kube.Client, manifests string) error {
 			ogvr = gvr.ValidatingWebhookConfiguration
 		}
 
-		_, err := kubeClient.Dynamic().Resource(ogvr).Namespace(obj.GetNamespace()).Patch(
+		// webhook failurePolicy is managed by istiod, so if currently we already have a webhook in cluster,
+		// we remove setting it from the manifest
+		webhooks, found, err := unstructured.NestedSlice(obj.Object, "webhooks")
+		if err != nil {
+			return fmt.Errorf("failed to get webhooks: %w", err)
+		}
+		if found && ogvr == gvr.ValidatingWebhookConfiguration {
```

Comment:
do we need to post-process this or can we just not include it in the webhook yaml?

Reply:
Oof, I thought the default was Ignore but it is Fail. So we cannot.

I am a bit concerned by this approach but don't see any obviously ideas...

TBH I wonder if we can just default to Fail now. The reason we didn't initially is we installed istio CRs during install. We don't anymore. But users may in their own pipelines...

@therealmitchconnors any ideas?

Reply:
So if we force we have a small possible missed validation. Not terrible (because it needs Istiod to be down as well to not work), but not great...

Reply:
I don't think this approach is great enough, and I can't come up with any ideas either... IIRC, last time when I fixed the readinessProbe issue, users may have been sensitive to the validation downtime; force to `Fail` changes the installation behavior. I think if post-processing and force setting to `Fail` are both not very great, the simplest approach is to keep the original `Ignore` and let pilot revise it?

Reply:
I kinda like the idea of directly setting it to `Fail`, since it's cleaner installation process and the legacy validation check can be removed. However, I'm not really sure how many users will be affected. Waiting for more ideas

Reply:
ok I think this approach may be fine but we shouldn't put the logic in `applyManifests` and should instead be in `ProcessDefaultWebhook`. 

This already fetches the webhook, so we don't need to do it again. It also fetches it in the typed form, so its much easier to process than all the casting. Its also a layering violation -- ideally `applyManifests` is agnostic to the inputs and just applies them

Reply:
Thanks, this makes sense to me.